### ISSUE TITLE
Markor translation & localization

### ISSUE THREAD

**Issue by gsantner** [gsantner/markor#20]
[![Translate - with Stringlate](https://img.shields.io/badge/stringlate-translate-green.svg)](https://lonamiwebs.github.io/stringlate/translate?git=https%3A%2F%2Fgithub.com%2Fgsantner%2Fmarkor.git&name=Markor&web=https%3A%2F%2Fgithub.com%2Fgsantner%2FMarkor) [![Chat - Matrix](https://img.shields.io/badge/chat-on%20matrix-blue.svg)](https://matrix.to/#/#markor:matrix.org) [![Chat - FreeNode IRC](https://img.shields.io/badge/chat-on%20irc-blue.svg)](https://kiwiirc.com/client/irc.freenode.net/?nick=markor-anon|?#markor)

This issue will be used to notify translators of the app. Please subscribe to this issue if you want to start translating or already translated something and want to keep the strings updated for some language.

~~You can use e.g. [Stringlate](https://f-droid.org/packages/io.github.lonamiwebs.stringlate/) to contribute translations - directly on your phone.~~ This app is now translated on Crowdin! Join the project [here](https://crowdin.com/project/markor/invite) (free of course) and start translating!

List of translators that are on GitHub and contributed at least once translations to one of my projects _(unified across gsantner projects)_:
* French - @milouse, @Poussinou, @Quent-in, @SansPseudoFix, @gbaudic, @romainrossi, @NicolasAzzopardi, @nailyk-fr, @rain79, @Primokorn, @Bridouz
* Polish - @kRkk, @pskosinski, @Bjornskjald, @bla-ke, @thesebas 
* Spanish - @larjona, @Lonami, @AlexandreInsua, @sidhenidon
* Italian - @simocosimo, @Daraf-IT, @matteoarrighi
* Russian - @xredor, @guland2000, @trubitsyn, @antonv6
* Brazilian - @luisfsr, @TiagoDanin, @mannuscritto, @RockyTV, @smarquespt, @Valeyard1
* Ukrainian - @cl0ne
* Swedish - @spitlo
* German - @gsantner
* Turkish - @mobile46
* Hindi - @kartikarora
* Czech - @vakabus, @Golbinex 
* Japanese - @naofum
* Sardinian - @asereze
* Greek - @nicolasmanic
* Dutch - @TopdRob
* Indonesian - @freedomofkeima
* Catalan - @Zagur
* Norwegian - @comradekingu
* Hungarian - @maxigaz
* Kabyle - @belkacem77
* Galacian - @xmgz
* Chinese (simp/cn) - @billchenchina
* Chineses (trad/tw) - @wlctw11
* Slovacisch - @marulinko

**Comment by gsantner** [comment]
Hi all, hacktober approaches :)

**Comment by gsantner** [comment]
Hi all, hacktober approaches :)

Register on <https://hacktoberfest.digitalocean.com/>, send a PR to update translations and get your t-shirt for free :).

**Comment by milouse** [comment]
Hi! Sorry for my silence, I was quiet busy these days. I hope to find time to finish french translation this week. The tshirt motivation may help ;)

**Comment by Abrasha** [comment]
Hi @gsantner! Would you like to translate your app to Ukrainian? I have already done it for [Stringlate](https://github.com/LonamiWebs/Stringlate/pull/147)

**Comment by gsantner** [comment]
go for it :D

**Comment by simocosimo** [comment]
Working on the Italian translation! :)

**Comment by gsantner** [comment]
@all Dear translators, I've put a lot of work over the weekend (plus today) into markor. there are a lot of new features, and pretty much everything is reworked. There are too a lot of new translateable words/strings available. If you have some time, markor would be pleased to have new features translated :) 👍 

Thanks!

**Comment by milouse** [comment]
Is it possible to have access to some kind of beta to better view in situation where the new strings are and what feature they describe? Maybe just put here a link to an apk? Thanks !

**Comment by gsantner** [comment]
@milouse  [markor-flavorGplay-release--v0.2.0.zip](https://github.com/gsantner/markor/files/1452060/markor-flavorGplay-release--v0.2.0.zip) or on gplay in some minutes. Feedback welcome! (non-translation stuff in issue 95)

**Comment by milouse** [comment]
Thanks for the apk, because I don't have play services on my phone.

**Comment by luisfsr** [comment]
Ops! I won't need to translate. Thanks @tiagodanin :)

**Comment by gsantner** [comment]
Thanks everybody for adding new translations! BTW: Forget to mention that `QuickNote` should not be translated. And Notebook/More/QuickNote is on bottom-layout, with low amount of space, so it should be short (you will see if it fits in your language)

**Comment by kartikarora** [comment]
I'll check that and send a patch if things are wrong, pretty soon

**Comment by kartikarora** [comment]
@gsantner Which string is `Quicknote`? Is it `R.string.pref_summary__quicknote_filepath`?

**Comment by gsantner** [comment]
QuickNote as standalone cannot be translated already, yeah it's about those strings where QuickNote is contained in the string, where the above said is one, yep

**Comment by kartikarora** [comment]
Things seem fine on my end!

**Comment by cl0ne** [comment]
@gsantner `section_format` and `title_activity_main` keys seem to be unused, what's their purpose?

**Comment by gsantner** [comment]
@cl0ne Neither of both are there in currennt version, if you use stringlate do an refresh

**Comment by gsantner** [comment]
hi everybody, I'm currently little working on settings and the translations. I will do an commit in some minutes. Can you please check the following if you have time?:

* Unless written in original english string - don't use the translation for `file` or `note`, use `document` instead
  * E.g. `Browse filesystem` should be still translated as `Browse filesystem`
  * But e.g. `Create new document` should **not** be translated to `Create new file`
  * Notebook can too be translated as Notebook
* Be sure to not translate `QuickNote` into local language, use that in your language too
* There are some new strings available to translate
* Strings that are currently outdated are tagged as `needed="false"` in the xml, you can skip these if you are directly editing the xml files 😄

**Comment by milouse** [comment]
Hi @gsantner you add a bunch of new string for the 2.5 version. One of them is just `name="none"`, which may be translated by *aucun* or *aucune* wether it refers to a male (e.g. « document » or « context » in french) or female subject (e.g. « catégorie » or « note »). Can you give us the context in which this new string will be used? (I think german translation may have the same need)

**Comment by gsantner** [comment]
For selecting priority. A, B, C, .. or None of them

![screenshot_20171209-122324](https://user-images.githubusercontent.com/6735650/33795153-e1537356-dcdb-11e7-93bb-b6e7c0cb6af3.png)

**Comment by gsantner** [comment]
Hi everybody!

I scripted a little as I wanted to know how good and much translated the app currently is. I included all languages that my apps were yet translated in. The first column says how much strings are already translated in the particular language with the second being more precise.

```
_______________________________________________________
 %     abs    code      language
_______________________________________________________
100%	(203)	de	German (Deutsch)
99%	(201)	tr	Turkish (Türkçe)
99%	(201)	fr	French (Français)
89%	(182)	zh-rTW	Chinese Traditional
88%	(180)	it	Italian (Italiano)
81%	(166)	ru	Russian (Русский)
81%	(166)	es	Spanish (Español)
78%	(159)	pl	Polish (Polski)
75%	(154)	sv	Swedish (Svenska)
73%	(149)	fa	Persian (فارسی)
66%	(136)	pt-rBR	Brazilian (Português, Brasil)
66%	(135)	zh-rCN	Chinese Simplified
65%	(132)	ja	Japanese
53%	(108)	uk	Ukrainian (Українська)
53%	(108)	kab	Kabyle (Taqbaylit)
53%	(108)	hi	Hindi
48%	(99)	nl	Dutch (Nederlands)
0%	(0)	sr	Serbian (Српски)
0%	(0)	sc	Sardinian (Sardinian)
0%	(0)	ro	Romanian (Română)
0%	(0)	pt-rPT	Portuguese (Português) / Portugal
0%	(0)	pt	Portuguese (Português) / Portugal
0%	(0)	ko-rKR	Korean
0%	(0)	in	Indonesian (Indonesia)
0%	(0)	eo	Esperanto (Esperanto)
0%	(0)	el	Greek (Ελληνικά)
0%	(0)	da	Danish (Dansk)
0%	(0)	cs	Czech (Čeština)
0%	(0)	ca	Catalan (Català)
0%	(0)	ar	Arabic (العربية)
```



I am happy about all translation update contributions : smile :

**Comment by gsantner** [comment]
Hi everybody!

I scripted a little as I wanted to know how good and much translated the app currently is. I included all languages that my apps were yet translated in. The first column says how much strings are already translated in the particular language with the second being more precise.

Especially right-to-left languages are missing currently. Here translation contributions and feedback about usability in RTL would be very helpful. 

```
Strings available for translation: 203                                                                                                                                                
_______________________________________________________
 %       abs    code      language
_______________________________________________________
100%	(203)	de	German (Deutsch)
99%	(201)	tr	Turkish (Türkçe)
99%	(201)	fr	French (Français)
89%	(182)	zh-rTW	Chinese Simplified
88%	(180)	it	Italian (Italiano)
81%	(166)	ru	Russian (Русский)
81%	(166)	es	Spanish (Español)
78%	(159)	pl	Polish (Polski)
75%	(154)	sv	Swedish (Svenska)
73%	(149)	fa	Persian (فارسی)
66%	(136)	pt-rBR	Brazilian (Português, Brasil)
66%	(135)	zh-rCN	Chinese Traditional
65%	(132)	ja	Japanese
53%	(108)	uk	Ukrainian (Українська)
53%	(108)	kab	Kabyle (Taqbaylit)
53%	(108)	hi	Hindi
48%	(99)	nl	Dutch (Nederlands)
0%	(0)	sr	Serbian (Српски)
0%	(0)	sc	Sardinian (Sardinian)
0%	(0)	ro	Romanian (Română)
0%	(0)	pt-rPT	Portuguese (Português) / Portugal
0%	(0)	ko-rKR	Korean
0%	(0)	in	Indonesian (Indonesia)
0%	(0)	eo	Esperanto (Esperanto)
0%	(0)	el	Greek (Ελληνικά)
0%	(0)	da	Danish (Dansk)
0%	(0)	cs	Czech (Čeština)
0%	(0)	ca	Catalan (Català)
0%	(0)	ar	Arabic (العربية)
```



I am happy about all translation update contributions : smile :

**Comment by comradekingu** [comment]
Norwegian Bokmål done https://github.com/gsantner/markor/pull/225
Language rework github.com/gsantner/markor/pull/226

**Comment by gsantner** [comment]
@comradekingu thanks! commented in PRs

**Comment by TiagoDanin** [comment]
PT-BR translation in progress :)

**Comment by billchenchina** [comment]
@gsantner well, something goes wrong..

`zh-rTW` means Chinese Traditional while `zh-rCN` stands for Chinese Simplified

**Comment by kartikarora** [comment]
hi-hi in progress

**Comment by gsantner** [comment]
@billchenchina 
No, its correct - for android resource system and matching. It had no other way up to Android 7. As minSdk is and wont go up so fast this wont likely change in near time. Its not about country here.

See below.

https://gist.github.com/amake/0ac7724681ac1c178c6f95a5b09f03ce

https://fabiohub.wordpress.com/2016/12/31/chinese-locale-in-android/

https://medium.com/fabiohub/chinese-locale-in-android-part-2-d1cfe87b3fb2

**Comment by RockyTV** [comment]
Why do you use Stringlate for translating? You can't use it in your desktop, requiring you to use your phone, and cloning each project to translate a few strings gets annoying after a few projects. Why not Crowdin or WebTransalteIt/Transifex?

**Comment by billchenchina** [comment]
@gsantner  right, well, but your table matches wrongly.

**Comment by gsantner** [comment]
@billchenchina ah you mean the percent table. updated to latest stats.

**Comment by gsantner** [comment]
## Hey everybody!

* I'm releasing Markor, dandelion and MemeTastic updates today (this means you get the download in 2-3 days on F-Droid)
* This is a major overhaul of translation IDs, and for markor some new features.
* If you have some time, please look carefully if everything is correctly translated into your language. If not, or something crashes, please report or update translation if possible! :)
* For all apps this is a published as a very minor release (v1.0.1 -> v1.0.2) without CHANGELOG ( Coffee  ). I will make a new point release (v1.0.2 -> v1.1.0) soon, probably beginning of next week. Depending on feedback (this will have changelog then).

**Notice**: Apart those languages with most translators (like German, French, Spanish) at most other languages many translations are missing! If you want to help translating the apps, theres much help available from project page, or inside app's about screen ;).


**Thanks!**

**Comment by gsantner** [comment]
For translators - install Stringlate from F-Droid, it allows **translation of apps directly from your phone.**
Just press the discover button and search for d*/ markor / memetastic.

 If already translated with it, its best to delete local translation repo, because of massive amount of recent changes.

**Cheers!**


https://f-droid.org/app/io.github.lonamiwebs.stringlate

**Comment by gsantner** [comment]
Hi everybody, there are some new features incoming with new strings soon :). Things like table of contents and Math formulas :).

I'm currently in slightly renaming hard-to-understand things, and have questions, what you think about it:
* Renaming TextModuleAction to Textaction, which gets translatable then. English non-translatable string is hard to guess
* Thinking of renaming `Preview` (the show rendered Markdown/todo button).  `Rendered View / Browse rendered text / Browse converted text` is not really good IMO. Preview seems to be somewhat wrong, because you already know what is it because you edit the text. What do you think fits best here? It's best to be short.

**Comment by asereze** [comment]
@gsantner It looks like there is a problem with some translations in Android 8.1 (LineageOS 15.1, I'm using a Xiaomi Redmi 4X). Every time I open dandelion or Cherry the app's language goes back to Italian (my system's language) even if I choose Sardinian in the settings. For some reason that doesn't happen with Stringlate.

I've seen that they got the same problem with VLC, and they solved it this way:
https://code.videolan.org/videolan/vlc-android/issues/612
I don't know if it could be useful.

**Comment by gsantner** [comment]
@asereze  Please make a seperate issue about that :)

**Comment by asereze** [comment]
@gsantner Done!

**Comment by gsantner** [comment]
Hi everybody!

I recently had much problems importing translations from Stringlate and I also saw a lot of outdated translation base (from old project state/version). Because of that I'm going back to a translation platform, the one I'm used to and was very happy with (in dandelion & Froody projects), which is Crowdin.

This means current original strings are always available, you can have discussion on meaning of a string and even talk to people translating the same project in the same language. 
This aims to improve 3 major points - me having less maintainence - translators having less todo and always get instantly submitted - and the users getting a good translation :+1: . 
**Thumbs up and thanks to every translator of the project, be it current or coming new :+1: .**

You can login/register here for my projects: <https://crowdin.com/profile/gsantner>. You only need one account and translate all my projects there. Existing translations (that you have done) is uploaded & in sync. Below are project specific short links if you want to use them:
* Markor - https://translate-markor.gsantner.net/
* dandelion - https://translate-dandelion.gsantner.net/
* MemeTastic - https://translate-memetastic.gsantner.net/

If a language is missing, just leave me a message and I will add it.

Below are the current translation stats:

![grafik](https://user-images.githubusercontent.com/6735650/46029320-5a566f00-c0f3-11e8-9d15-a7d822c1fe05.png)

**Comment by TiagoDanin** [comment]
Amazing!! I love Crowdin.
@gsantner Auto Memory is enabled?

**Comment by TiagoDanin** [comment]
A workflow to apply automatically "Translation Memory".
Crowdin ~> [Workflows](https://crowdin.com/resources#workflows) ~> Create a new workflow.
![image](https://user-images.githubusercontent.com/5731176/46041778-072bee80-c0ea-11e8-97e8-ba4c54a7d5ec.png)
![image](https://user-images.githubusercontent.com/5731176/46041854-36daf680-c0ea-11e8-8e6d-57a676748f72.png)

**Comment by gsantner** [comment]
@TiagoDanin Have enabled that above, and now have triggered TM restore on all projects. Across all 4 projects now there are 600 translated based off TM across all languages.

**Comment by gsantner** [comment]
Update is out!

Check the out the news, on the projects biggest news/blog post ever!
<https://gsantner.net/blog/2018/12/09/markor-release-v1.5.html?source=github>

**Comment by ema-pe** [comment]
Hi, I'm leaving as Italian translator. I'll continue to contribute closely until **20 April 2019**, after this date maybe I'll contribute sporadically, but I won't follow the translation status anymore. I do this because I don't have time to follow too many projects, and I can't keep the quality high.

Fortunately there are other italian translators active, like Daraf or .mau. on Crowdin.

This action is related only to localization. I'm a daily user of Markor, so if I found any bug or problems I'll open an issue on GitHub.

Have a good day!
Emanuele

**Comment by comradekingu** [comment]
I have left Crowdin too.

**Comment by gsantner** [comment]
Hi translators.

Thanks for your done help or your future help. Everybody is free to  join & help or stop to do so. Thanks for letting know, but as everybody is doing this voluntary everybody is can go, leave, go on whenever wanted. Everbody has own priorities, be it family or just other projects.

**Comment by billchenchina** [comment]
Me: preparing for NCEE(https://en.m.wikipedia.org/wiki/National_Higher_Education_Entrance_Examination), may continue translate for zh_CN after examination

**Comment by linsui** [comment]
Could you please add more strings? Some strings in https://github.com/gsantner/markor/blob/master/app/src/main/res/values/string-not_translatable.xml is translatable. And the [Help page](https://github.com/gsantner/markor/blob/c60a4145f9f15e8a113038de62e76f53faa3804b/app/src/main/res/raw/resources.md) is translatable, too. The language seletor should be translatable, but I can't find it.

**Comment by gsantner** [comment]
@linsui 
All strings that need translation are in strings.xml, ignore -not_translatable (hence the name).

On language selection, the language names/codes itself are coming from java itself. As so many languages are out and stuff is already translated in java itself by locale this doesnt need additionally to be done by markor.

**Comment by mvigneshtuty** [comment]
@gsantner 
Could you please add Tamil language to Crowdin. I can contribute to do translations.

**Comment by homocomputeris** [comment]
@gsantner, there are strings do not appear at Crowdin and are not translated in the UI (2.2.6 F-Droid):

1. Settings - General
- Features
- Launcher 
- App drawer launchers...
- Debug
2. Settings - View mode
- Inject
- Text
3. Settings - Theme
- Auto
3. More menu
- Austrian developer...
- Community
- Join the community
- Copy build information

**Comment by gsantner** [comment]
I browsed through the strings and right now all strings **that I want to be translated** are marked as such.

**Comment by gsantner** [comment]
Hello everybody,

I wanted to know how you feel about the GitHub discussions feature used in the Markor project.  
So for some background why I started using it in the project is because of the high mixture of questions/discussion, feature requests and bug reports. Overall the issue is that people tend to ignore the template for creating bug reports and just write freely. And very often it is the case, the issue is specific to already fixed/older versions, or it's specific to some Android version. Which is often missing information. So then I started a separation between real bug reports, where fields have to be filled out - and general discussions & feature requests (GitHub discussions). Two things discussions have that the issue tracker doesn't  are specific highlighted answers, and threaded discussions.

By now totally 5 pages of discussions amassed, which is a lot (and also a lot similar duplicate questions).

![grafik](https://user-images.githubusercontent.com/6735650/185814376-9ef6fc1a-f3d8-42dd-ba02-297bb6fd0a5f.png)

I wanted to know, how you feel about it - should we maybe drop the GitHub discussions "experiment"? keep the split?   
Why I'm asking this? Markor takes a lot of time in terms of maintainence (which includes writing responses). And I would rather do more for the code and improve the project than writing off & closing duplicates & already fixed issues 😄 .  I.e. could really need some help on maintaining the issue & discussion tracker - and improving the way issues/discussions/faq are used.

After all, either way I as maintainer answer most of questions/issues/feature-requests - and the most popular way of asking is sadly still sending me a private mail, whereas I explicitly not write my email in the app & ask people to refrain from mailing --  suggesting instead to ask publicly, resulting also in the question & answer getting public knowledge instead being part of some private mail.
